spec/models/reservation_spec.rb

```diff
@@ -21,6 +21,72 @@
   # can't run if nil (?)
   #
 
+  describe 'counter cache' do
+    context 'newly overdue reservation' do
+      it 'increments when a reservation is marked as overdue' do
+        model = FactoryGirl.build_stubbed(:equipment_model)
+        allow(EquipmentModel).to receive(:find).with(model.id).and_return(model)
+        res = FactoryGirl.create(:checked_out_reservation,
+                                 equipment_model: model,
+                                 start_date: Time.zone.today - 2.days,
+                                 due_date: Time.zone.today - 1.day)
+        res.update_columns(overdue: false)
+        expect(model).to receive(:increment).with(:overdue_count).once
```

Comment:
The eq model mock here is a FactoryGirl mock, not a spy, because it needs to pass the reservation validations